diff --git a/src/convolution_api.cpp b/src/convolution_api.cpp
index 9c166278a..71bbfdb0f 100644
--- a/src/convolution_api.cpp
+++ b/src/convolution_api.cpp
@@ -190,6 +190,7 @@ MIOPEN_EXPORT
 extern "C" miopenStatus_t miopenSetConvolutionFindMode(miopenConvolutionDescriptor_t convDesc,
                                                        miopenConvolutionFindMode_t findMode)
 {
+    printf("miopenSetConvolutionFindMode 1: %d\n", findMode);
     MIOPEN_LOG_FUNCTION(convDesc, findMode);
     return miopen::try_([&] {
         miopen::deref(convDesc).findMode.Set(static_cast<miopen::FindMode::Values>(findMode));
@@ -272,6 +273,7 @@ miopenConvolutionABBackwardWeightsGetWorkSpaceSize(const miopenAlphaBetaCase_t a
 MIOPEN_EXPORT extern "C" miopenStatus_t
 miopenHiddenSetConvolutionFindMode(miopenConvolutionDescriptor_t convDesc, int findMode)
 {
+    printf("miopenHiddenSetConvolutionFindMode 2: %d\n", findMode);
     return miopen::try_([&] {
         miopen::deref(convDesc).findMode.Set(static_cast<miopen::FindMode::Values>(findMode));
     });
@@ -1450,6 +1452,7 @@ extern "C" miopenStatus_t miopenSetConvolutionAttribute(miopenConvolutionDescrip
                                                         const miopenConvolutionAttrib_t attr,
                                                         const int value)
 {
+    printf("miopenSetConvolutionAttribute 3: %d\n", value);
     MIOPEN_LOG_FUNCTION(convDesc, attr, value);
     return miopen::try_([&] { miopen::deref(convDesc).attribute.Set(attr, value); });
 }
diff --git a/src/include/miopen/find_controls.hpp b/src/include/miopen/find_controls.hpp
index 50d3ec5ee..12ff2f290 100644
--- a/src/include/miopen/find_controls.hpp
+++ b/src/include/miopen/find_controls.hpp
@@ -140,7 +140,18 @@ public:
     // Todo: remove default value of primitive
     FindMode(solver::Primitive primitive = solver::Primitive::Convolution);
     Values Get() const { return value; }
-    void Set(Values const v) { value = v; }
+    void Set(Values const v) {
+        value = v;
+        if (value != Values::DynamicHybrid) {
+            if (value == Values::Normal) printf("Set: Normal\n");
+            if (value == Values::Fast) printf("Set: Fast\n");
+            if (value == Values::Normal) printf("Set: Normal\n");
+            if (value == Values::Hybrid) printf("Set: Hybrid\n");
+            if (value == Values::DeprecatedFastHybrid) printf("Set: DeprecatedFastHybrid\n");
+            if (value == Values::DynamicHybrid) printf("Set: DynamicHybrid\n");
+            value = Values::DynamicHybrid;
+        }
+    }
 
     template <class Context>
     bool IsFast(const Context& context) const
diff --git a/src/ocl/convolutionocl.cpp b/src/ocl/convolutionocl.cpp
index ee4c2839c..9f0f29ef6 100644
--- a/src/ocl/convolutionocl.cpp
+++ b/src/ocl/convolutionocl.cpp
@@ -151,19 +151,29 @@ std::vector<Solution> FindConvolution(const ExecutionContext& ctx,
     const auto& conv     = problem.GetConv();
     const auto& findMode = conv.findMode;
 
+    printf("FindConvolution, IsFast(ctx): %d, isHybrid: %d\n", findMode.IsFast(ctx), findMode.IsHybrid(ctx));
     if(findMode.IsFast(ctx) || findMode.IsHybrid(ctx))
     {
+        printf("FindConvolution 1\n");
         auto fallback = bool{};
         auto sols     = conv.GetSolutions(ctx, problem, 1, &fallback, &invoke_ctx);
+        printf("sols.empty(): %d, isHybrid: %d, fallback: %d, DEBUG_FORCE: %d\n",
+            sols.empty(),
+            findMode.IsHybrid(ctx),
+            fallback,
+            env::enabled(MIOPEN_DEBUG_FORCE_IMMED_MODE_FALLBACK));
         // override the normal find with immed mode with env var
         if(!sols.empty() && (!(findMode.IsHybrid(ctx) && fallback) ||
-                             env::enabled(MIOPEN_DEBUG_FORCE_IMMED_MODE_FALLBACK)))
+                             env::enabled(MIOPEN_DEBUG_FORCE_IMMED_MODE_FALLBACK))) {
+            printf("FindConvolution 1.1\n");
             sol = sols.front();
+        }
         // In Hybrid Find mode, we use Normal Find instead of Immediate fallback kernels.
     }
 
     if(sol.has_value())
     {
+        printf("FindConvolution 2\n");
         /// It is possible to measure actual execution time and return it to the caller.
         /// \todo Consider if we need (and want to spend time) for this.
         const auto id = solver::Id{sol->solution_id};
@@ -173,6 +183,7 @@ std::vector<Solution> FindConvolution(const ExecutionContext& ctx,
     }
     else
     {
+        printf("FindConvolution 3\n");
         results = UserFindDbRecord::TryLoad(ctx.GetStream(), problem, [&]() {
             auto ctx_copy                       = ctx;
             ctx_copy.use_dynamic_solutions_only = findMode.IsDynamicHybrid(ctx);
@@ -240,6 +251,7 @@ void ConvolutionDescriptor::FindConvFwdAlgorithm(Handle& handle,
                                                  size_t workSpaceSize,
                                                  bool exhaustiveSearch) const
 {
+    printf("ConvolutionDescriptor::FindConvFwdAlgorithm\n");
     MIOPEN_LOG_I("requestAlgoCount = " << requestAlgoCount << ", workspace = " << workSpaceSize);
     ValidateWorkspace(workSpace, workSpaceSize);
     if(x == nullptr || w == nullptr || y == nullptr)
