From d10e8be1b60f23c20bd101743d2b220dfb57d3bc Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 25 Jul 2024 14:41:14 -0700
Subject: [PATCH 1/5] more gfx devices to default CMakeLists.txt

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12efe8f..f9a4d7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,10 +35,14 @@ set(DEFAULT_GPUS
       gfx940
       gfx941
       gfx942
+      gfx1010
       gfx1030
+      gfx1035
+      gfx1036
       gfx1100
       gfx1101
-      gfx1102)
+      gfx1102
+      gfx1103)
 
 # Load CMake modules
 #==================================================================================================
-- 
2.41.1

