From fbec847c285080c07c8d5bb6a49ff49196438b42 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 6 Jan 2024 01:29:45 +0200
Subject: [PATCH] gfx1010, 1030, 1035 and 1036

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba4bd72..d17a9c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,19 +26,10 @@ option(TRACE                                   "Enable additional tracing"
 # Default GPU architectures to build
 #==================================================================================================
 set(DEFAULT_GPUS
-      gfx803
-      gfx900:xnack-
-      gfx906:xnack-
-      gfx908:xnack-
-      gfx90a:xnack-
-      gfx90a:xnack+
-      gfx940
-      gfx941
-      gfx942
+      gfx1010
       gfx1030
-      gfx1100
-      gfx1101
-      gfx1102)
+      gfx1035
+      gfx1036)
 
 # Load CMake modules
 #==================================================================================================
-- 
2.41.0

