From 6aa764c3663d6f1ccabb04413305447d6a80d062 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Tue, 19 Dec 2023 15:16:58 -0800
Subject: [PATCH 1/2] by default no 3RD_DWORD support for not listed gpus

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 include/ck/ck.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/ck/ck.hpp b/include/ck/ck.hpp
index 069ff7fc..77b52171 100644
--- a/include/ck/ck.hpp
+++ b/include/ck/ck.hpp
@@ -55,6 +55,8 @@
 #define CK_BUFFER_RESOURCE_3RD_DWORD 0x31014000
 #elif defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) // for GPU code
 #define CK_BUFFER_RESOURCE_3RD_DWORD 0x31004000
+#else
+#define CK_BUFFER_RESOURCE_3RD_DWORD -1
 #endif
 
 // FMA instruction
-- 
2.41.0

