From 9ed3db4736aaeb1fd556fe14739fa198e3988df2 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 18 May 2024 18:39:43 -0700
Subject: [PATCH 5/6] fallback support debug patch

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 tensilelite/Tensile/TensileCreateLibrary.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tensilelite/Tensile/TensileCreateLibrary.py b/tensilelite/Tensile/TensileCreateLibrary.py
index d2ac2166..ce6c0d5f 100644
--- a/tensilelite/Tensile/TensileCreateLibrary.py
+++ b/tensilelite/Tensile/TensileCreateLibrary.py
@@ -1343,7 +1343,8 @@ def TensileCreateLibrary():
     if arch in architectureMap:
       logicArchs.add(architectureMap[arch])
     else:
-      printExit("Architecture %s not supported" % arch)
+      printl("Architecture %s not supported" % arch)
+      printExit("ArchitectureMap: %s" % architectureMap)
 
   if globalParameters["LazyLibraryLoading"] and not (globalParameters["MergeFiles"] and globalParameters["SeparateArchitectures"]):
     printExit("--lazy-library-loading requires --merge-files and --separate-architectures enabled")
-- 
2.41.0

