From f1b12565107005f62c5ef0f47cd18a23cbcc0b1e Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 7 Nov 2024 22:24:08 -0800
Subject: [PATCH 11/12] allow to specify target gpu list

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 build_rocm.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build_rocm.sh b/build_rocm.sh
index 18c9cfaa046..86ac0cb005e 100755
--- a/build_rocm.sh
+++ b/build_rocm.sh
@@ -16,6 +16,12 @@ else
     rocm_version_str=${2}
     echo "Using rocm version parameter: ${rocm_version_str}"
 fi
+if [ ! -z "$3" ]; then
+    export PYTORCH_ROCM_ARCH="$3"
+    echo "Building Pytorch for GPUs: ${PYTORCH_ROCM_ARCH}"
+else
+	echo "Building Pytorch for GPU's used on rocm sdk build."
+fi
 unset LDFLAGS
 export CFLAGS="-Wno-error=maybe-uninitialized"
 unset CPPFLAGS
-- 
2.48.1

