From e7bd9c1228867440ffb6a2252a84ab957f4b9c43 Mon Sep 17 00:00:00 2001
From: Pruthvi Madugundu <pruthvigithub@gmail.com>
Date: Mon, 16 Sep 2024 10:14:27 -0500
Subject: [PATCH 14/40] Enable AOTriton kernel compression

---
 cmake/External/aotriton.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/External/aotriton.cmake b/cmake/External/aotriton.cmake
index cb6080f5f7a..8c2984ae9e9 100644
--- a/cmake/External/aotriton.cmake
+++ b/cmake/External/aotriton.cmake
@@ -19,7 +19,6 @@ if(NOT __AOTRITON_INCLUDED)
       BINARY_DIR ${__AOTRITON_BUILD_DIR}
       PREFIX ${__AOTRITON_INSTALL_DIR}
       CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${__AOTRITON_INSTALL_DIR}
-      -DAOTRITON_COMPRESS_KERNEL=OFF
       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
       -DAOTRITON_NO_PYTHON=ON
       -DAOTRITON_NO_SHARED=OFF
-- 
2.43.0

