From 1bcf9b817efcd469865ab3ec76ef7cf671800967 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 18 May 2024 19:38:33 -0700
Subject: [PATCH 2/3] rocm_sdk specific llvm tool paths

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install.sh b/install.sh
index 6b7857a..9c2713a 100755
--- a/install.sh
+++ b/install.sh
@@ -497,7 +497,7 @@ cmake_executable=cmake
 # If user provides custom ${rocm_path} path for hcc it has lesser priority,
 # but with hip-clang existing path has lesser priority to avoid use of installed clang++
 if [[ "${build_hip_clang}" == true ]]; then
-  export PATH=${rocm_path}/bin:${rocm_path}/hip/bin:${rocm_path}/llvm/bin:${PATH}
+  export PATH=${rocm_path}/bin:${rocm_path}/hip/bin:${rocm_path}/bin:${PATH}
 fi
 
 # #################################################
-- 
2.41.0

