From 7ce89c57981e13050bdf33d6712e029800e6ecc5 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 27 Jul 2024 01:08:15 -0700
Subject: [PATCH 1/2] offload arch gpu list update

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 config/m4/rocm.m4 | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/config/m4/rocm.m4 b/config/m4/rocm.m4
index 8a2bdc2..fc90e25 100644
--- a/config/m4/rocm.m4
+++ b/config/m4/rocm.m4
@@ -10,10 +10,22 @@ ROCM_ARCH90A="--offload-arch=gfx90a"
 ROCM_ARCH94="--offload-arch=gfx940 \
 --offload-arch=gfx941 \
 --offload-arch=gfx942"
-ROCM_ARCH10="--offload-arch=gfx1030"
+ROCM_ARCH10="--offload-arch=gfx1010 \
+--offload-arch=gfx1011 \
+--offload-arch=gfx1012 \
+--offload-arch=gfx1030 \
+--offload-arch=gfx1031 \
+--offload-arch=gfx1032 \
+--offload-arch=gfx1035 \
+--offload-arch=gfx1036"
 ROCM_ARCH11="--offload-arch=gfx1100 \
 --offload-arch=gfx1101 \
---offload-arch=gfx1102"
+--offload-arch=gfx1102 \
+--offload-arch=gfx1103 \
+--offload-arch=gfx1150 \
+--offload-arch=gfx1151"
+ROCM_ARCH12="--offload-arch=gfx1200 \
+--offload-arch=gfx1201"
 
 # ROCM_PARSE_FLAGS(ARG, VAR_LIBS, VAR_LDFLAGS, VAR_CPPFLAGS)
 # ----------------------------------------------------------
@@ -138,9 +150,9 @@ AS_IF([test "x$with_rocm" != "xno"],
 
     AS_IF([test "x$rocm_happy" = "xyes"],
         [AS_IF([test "x$with_rocm_arch" = "xall"],
-          [ROCM_ARCH="${ROCM_ARCH908} ${ROCM_ARCH90A} ${ROCM_ARCH94} ${ROCM_ARCH10} ${ROCM_ARCH11} ${ROCM_ARCH_NATIVE}"],
+          [ROCM_ARCH="${ROCM_ARCH908} ${ROCM_ARCH90A} ${ROCM_ARCH94} ${ROCM_ARCH10} ${ROCM_ARCH11} ${ROCM_ARCH12} ${ROCM_ARCH_NATIVE}"],
         [AS_IF([test "x$with_rocm_arch" = "xall-arch-no-native"],
-          [ROCM_ARCH="${ROCM_ARCH908} ${ROCM_ARCH90A} ${ROCM_ARCH94} ${ROCM_ARCH10} ${ROCM_ARCH11}"],
+          [ROCM_ARCH="${ROCM_ARCH908} ${ROCM_ARCH90A} ${ROCM_ARCH94} ${ROCM_ARCH10} ${ROCM_ARCH11} ${ROCM_ARCH12}"],
         [ROCM_ARCH="$with_rocm_arch"])])
         AS_IF([test "$ROCM_VERSION_60_OR_GREATER" = "1"],
           AC_SUBST([ROCM_ARCH], ["$ROCM_ARCH"]),
-- 
2.48.1

