From bc3fa94e64710d8aae5078a4931049df5aa73ab6 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 6 Jan 2024 09:21:38 +0200
Subject: [PATCH] rocWMMA disable tests with build error

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1682695..4bdcda6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 # Top level configs
 if( CMAKE_PROJECT_NAME STREQUAL "rocwmma" )
-  option( ROCWMMA_BUILD_TESTS "Build rocWMMA tests" ON )
+  option( ROCWMMA_BUILD_TESTS "Build rocWMMA tests" OFF )
   option( ROCWMMA_BUILD_SAMPLES "Build rocWMMA samples" ON )
   option( ROCWMMA_BUILD_ASSEMBLY "Output assembly files" OFF )
 endif()
@@ -80,7 +80,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_DIR}/library/include/rocwmma/internal/roc
                "${CMAKE_CURRENT_SOURCE_DIR}/library/include/rocwmma/rocwmma-version.hpp" )
 
 rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
-  TARGETS "gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102" )
+  TARGETS "gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1030;gfx1035;gfx1100;gfx1101;gfx1102" )
 
 # Variable AMDGPU_TARGET must be a cached variable and must be specified before calling find_package(hip)
 # This is because hip-config.cmake sets --offload-arch via AMDGPU_TARGET cached variable __after__ setting
-- 
2.41.0

