From ff9897b3a7682f18395e290a0b6b7e9ddbf00001 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Fri, 3 May 2024 14:21:04 -0700
Subject: [PATCH 1/4] disable tests which fails to build

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f546d84..912e3b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,7 +46,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fclang-abi-compat=17")
 
 # Top level configs
 if( CMAKE_PROJECT_NAME STREQUAL "rocwmma" )
-  option( ROCWMMA_BUILD_TESTS "Build rocWMMA tests" ON )
+  option( ROCWMMA_BUILD_TESTS "Build rocWMMA tests" OFF )
   option( ROCWMMA_BUILD_SAMPLES "Build rocWMMA samples" ON )
   option( ROCWMMA_BUILD_ASSEMBLY "Output assembly files" OFF )
 endif()
-- 
2.43.0

