From e9b4bdfd3ef99b74c2298a8f36baa898f2f726a9 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 18 May 2024 18:30:45 -0700
Subject: [PATCH] add gfx1035 to default build targets

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8dc6181..7fc18c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,6 +171,7 @@ if(NOT DEFINED AMDGPU_TARGETS)
     gfx908:xnack-
     gfx1010
     gfx1030
+    gfx1035
     ${OPTIONAL_AMDGPU_TARGETS}
   )
 endif()
-- 
2.41.0

