From c44148d15fbdc7a0be8374d3ebd8e79cee176f59 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 4 Nov 2024 18:50:50 -0800
Subject: [PATCH 5/6] add pybind11 to requirements.txt

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 requirements.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/requirements.txt b/requirements.txt
index 0abbfe0..da69496 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -5,3 +5,4 @@ pluggy
 numpy
 setuptools
 wheel
+pybind11
-- 
2.43.0

