From 49b5a61480557a97bdeee7a4817d3b2a4c599c51 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 18 May 2024 18:17:42 -0700
Subject: [PATCH 01/14] add mageia 9 support to install.sh

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 install.sh | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/install.sh b/install.sh
index fc644b87..46c95775 100755
--- a/install.sh
+++ b/install.sh
@@ -19,10 +19,10 @@ supported_distro( )
   fi
 
   case "${ID}" in
-    ubuntu|centos|rhel|fedora|sles|opensuse-leap)
+    ubuntu|centos|rhel|fedora|sles|opensuse-leap|mageia)
         true
         ;;
-    *)  printf "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, and Fedora\n"
+    *)  printf "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, Fedora and Mageia\n"
         exit 2
         ;;
   esac
@@ -220,11 +220,16 @@ install_packages( )
       install_dnf_packages "${library_dependencies_fedora[@]}"
       ;;
 
+    mageia)
+#     elevate_if_not_root dnf -y update
+      #install_dnf_packages "${library_dependencies_fedora[@]}"
+      ;;
+
     sles|opensuse-leap)
        install_zypper_packages "${library_dependencies_sles[@]}"
         ;;
     *)
-      echo "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, and Fedora"
+      echo "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, Fedora and Mageia"
       exit 2
       ;;
   esac
@@ -548,6 +553,9 @@ if [[ "${install_package}" == true ]]; then
     fedora)
       elevate_if_not_root dnf install rocblas-*.rpm
     ;;
+    mageia)
+    ;;
+
     sles|opensuse-leap)
       elevate_if_not_root zypper -n --no-gpg-checks install rocblas-*.rpm
     ;;
-- 
2.43.0

