From 861474c220a0697af7d72e26e762a86ae250882f Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sun, 24 Nov 2024 22:48:12 -0800
Subject: [PATCH 6/7] vega gpu improvements

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 src/target_properties.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/target_properties.cpp b/src/target_properties.cpp
index a330a3e52..2826d21b7 100644
--- a/src/target_properties.cpp
+++ b/src/target_properties.cpp
@@ -52,6 +52,9 @@ static std::string GetDeviceNameFromMap(const std::string& in)
         {"gfx804", "gfx803"},
         {"Vega10", "gfx900"},
         {"gfx901", "gfx900"},
+        {"vega10", "gfx900"},
+        {"vega11", "gfx902"},
+        {"vega20", "gfx906"},
         {"navi10", "gfx1010"},
         {"navi12", "gfx1011"},
         {"navi14", "gfx1012"},
-- 
2.41.1

