From dfd5cf9e04946592514e26f77d59475136f27a92 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Fri, 26 Jan 2024 01:59:17 -0800
Subject: [PATCH 2/2] rocm 6.0.0 hipsparse directory location update

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 sparse_hip/include/magmasparse_types.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sparse_hip/include/magmasparse_types.h b/sparse_hip/include/magmasparse_types.h
index be8b7c217..35a72b6bd 100644
--- a/sparse_hip/include/magmasparse_types.h
+++ b/sparse_hip/include/magmasparse_types.h
@@ -21,7 +21,7 @@
 #endif
 
 // includes CUDA
-#include <hipsparse.h>
+#include <hipsparse/hipsparse.h>
 
 /* (author: Cade Brown <cbrow216@vols.utk.edu>
  *
-- 
2.41.0

