From 91ace67bc4898c0148a32849bb0eee63d64a4c5a Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 6 Jun 2024 10:51:14 -0700
Subject: [PATCH 3/3] search library both from the lib64 and lib

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 python_smi_tools/rsmiBindings.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/python_smi_tools/rsmiBindings.py b/python_smi_tools/rsmiBindings.py
index 3cd2e9f..0ae20c7 100644
--- a/python_smi_tools/rsmiBindings.py
+++ b/python_smi_tools/rsmiBindings.py
@@ -26,7 +26,9 @@ def initRsmiBindings(silent=False):
     if (rocm_smi_lib_path != None):
         path_librocm = rocm_smi_lib_path
     else:
-        path_librocm = os.path.dirname(os.path.realpath(__file__)) + '/../../lib/librocm_smi64.so.7'
+        path_librocm = os.path.dirname(os.path.realpath(__file__)) + '/../../lib64/librocm_smi64.so.7'
+        if not os.path.isfile(path_librocm):
+            path_librocm = os.path.dirname(os.path.realpath(__file__)) + '/../../lib/librocm_smi64.so.7'
 
     if not os.path.isfile(path_librocm):
         print_silent('Unable to find %s . Trying /opt/rocm*' % path_librocm)
-- 
2.41.1

