From 21bc9643a47de2215f295a8b33f04a5cd75ac787 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sun, 29 Sep 2024 21:21:11 -0700
Subject: [PATCH 2/3] printout pidfile location

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 src/common/read_config.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/common/read_config.c b/src/common/read_config.c
index 8714e544d0..e282ecf02d 100644
--- a/src/common/read_config.c
+++ b/src/common/read_config.c
@@ -5182,8 +5182,11 @@ static int _validate_and_set_defaults(slurm_conf_t *conf,
 		conf->slurmctld_debug = LOG_LEVEL_INFO;
 
 	if (!s_p_get_string(&conf->slurmctld_pidfile,
-			    "SlurmctldPidFile", hashtbl))
+                            "SlurmctldPidFile", hashtbl)) {
+                printf("using default pidfile");
 		conf->slurmctld_pidfile = xstrdup(DEFAULT_SLURMCTLD_PIDFILE);
+        }
+        printf("pidfile: %s\n", conf->slurmctld_pidfile);
 
 	(void )s_p_get_string(&conf->slurmctld_logfile, "SlurmctldLogFile",
 			      hashtbl);
-- 
2.41.1

