From db50586bcf4c48a317de3b8285ad3de286d9b0cc Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 20 Jan 2024 10:50:47 -0800
Subject: [PATCH] rocBLAS mageia fixes

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt |  6 +++---
 install.sh     | 14 +++++++++++---
 2 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 655b31ea..8516b928 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,9 +108,9 @@ rocm_setup_version( VERSION ${VERSION_STRING} )
 list( APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/llvm ${ROCM_PATH} ${ROCM_PATH}/hip /opt/rocm/llvm /opt/rocm /opt/rocm/hip )
 
 # setting target list based on ROCm version
-set( TARGET_LIST_ROCM_5.6 "gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102")
-set( TARGET_LIST_ROCM_5.7 "gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102")
-set( TARGET_LIST_ROCM_6.0 "gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102")
+set( TARGET_LIST_ROCM_5.6 "gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx1010;gfx1012;gfx1030;gfx1035;gfx1100;gfx1101;gfx1102")
+set( TARGET_LIST_ROCM_5.7 "gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1035;gfx1100;gfx1101;gfx1102")
+set( TARGET_LIST_ROCM_6.0 "gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1035;gfx1100;gfx1101;gfx1102")
 
 if(ROCM_PLATFORM_VERSION)
   if(${ROCM_PLATFORM_VERSION} VERSION_LESS 5.7.0)
diff --git a/install.sh b/install.sh
index fc644b87..46c95775 100755
--- a/install.sh
+++ b/install.sh
@@ -19,10 +19,10 @@ supported_distro( )
   fi
 
   case "${ID}" in
-    ubuntu|centos|rhel|fedora|sles|opensuse-leap)
+    ubuntu|centos|rhel|fedora|sles|opensuse-leap|mageia)
         true
         ;;
-    *)  printf "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, and Fedora\n"
+    *)  printf "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, Fedora and Mageia\n"
         exit 2
         ;;
   esac
@@ -220,11 +220,16 @@ install_packages( )
       install_dnf_packages "${library_dependencies_fedora[@]}"
       ;;
 
+    mageia)
+#     elevate_if_not_root dnf -y update
+      #install_dnf_packages "${library_dependencies_fedora[@]}"
+      ;;
+
     sles|opensuse-leap)
        install_zypper_packages "${library_dependencies_sles[@]}"
         ;;
     *)
-      echo "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, and Fedora"
+      echo "This script is currently supported on Ubuntu, CentOS, RHEL, SLES, OpenSUSE-Leap, Fedora and Mageia"
       exit 2
       ;;
   esac
@@ -548,6 +553,9 @@ if [[ "${install_package}" == true ]]; then
     fedora)
       elevate_if_not_root dnf install rocblas-*.rpm
     ;;
+    mageia)
+    ;;
+
     sles|opensuse-leap)
       elevate_if_not_root zypper -n --no-gpg-checks install rocblas-*.rpm
     ;;
-- 
2.41.0

