From 40df60d6544a9ed48f820eb7eaf7b1e28a3c15f1 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 15 Mar 2025 22:16:09 -0700
Subject: [PATCH 3/4] cmakelist.txt gfx1150 fix?

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c062d1dd3..e527a40d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,6 @@ endif()
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
 if (DTYPES)
-    add_definitions(-DDTYPES)
     if (DTYPES MATCHES "int8")
         add_definitions(-DCK_ENABLE_INT8)
         set(CK_ENABLE_INT8 "ON")
-- 
2.48.1

