From c784f628cd2705b6f49d7b801f9e2506760d6e26 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Tue, 19 Dec 2023 15:16:58 -0800
Subject: [PATCH 1/6] no 3RD_DWORD support by default for GPUs

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 include/ck/ck.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/ck/ck.hpp b/include/ck/ck.hpp
index 88efb0277..da5bb86c6 100644
--- a/include/ck/ck.hpp
+++ b/include/ck/ck.hpp
@@ -55,6 +55,8 @@
 #define CK_BUFFER_RESOURCE_3RD_DWORD 0x31014000
 #elif defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) // for GPU code
 #define CK_BUFFER_RESOURCE_3RD_DWORD 0x31004000
+#else
+#define CK_BUFFER_RESOURCE_3RD_DWORD -1
 #endif
 
 // FMA instruction
-- 
2.41.1

