From 2fba4b9405e2842897f5da473b9bba5df8a25b0e Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Fri, 3 May 2024 13:04:15 -0700
Subject: [PATCH] disable DPP from gfx1035 1036 and 1103

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 rocprim/include/rocprim/config.hpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/rocprim/include/rocprim/config.hpp b/rocprim/include/rocprim/config.hpp
index 22edf611..60c44302 100644
--- a/rocprim/include/rocprim/config.hpp
+++ b/rocprim/include/rocprim/config.hpp
@@ -76,6 +76,13 @@
     #define ROCPRIM_DETAIL_HAS_DPP 1
 #endif
 
+#if defined(__gfx1035__) || defined(__gfx1036__) || defined(__gfx1103__)
+        #ifndef ROCPRIM_DISABLE_DPP
+                #define ROCPRIM_DISABLE_DPP
+                #define ROCPRIM_DETAIL_USE_DPP 0
+        #endif
+#endif
+
 #if !defined(ROCPRIM_DISABLE_DPP) && defined(ROCPRIM_DETAIL_HAS_DPP)
     #define ROCPRIM_DETAIL_USE_DPP 1
 #else
-- 
2.48.1

