From 82633ab4779054e0a165daec14f43fd9acedf960 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 20 May 2024 20:19:29 -0700
Subject: [PATCH] discussion link

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 instructions.txt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 instructions.txt

diff --git a/instructions.txt b/instructions.txt
new file mode 100644
index 0000000..f20417e
--- /dev/null
+++ b/instructions.txt
@@ -0,0 +1 @@
+https://github.com/ROCm/ROCm/issues/3132
-- 
2.41.0

